modules/subnet/subnet.tf

```diff
@@ -5,11 +5,16 @@ locals {
   dhcp_default_options = data.oci_core_dhcp_options.dhcp_options.options.0.id
 }
 
+data "oci_identity_availability_domains" "ads" {
+  compartment_id = var.tenancy_ocid
+}
+
 resource "oci_core_subnet" "vcn_subnet" {
   for_each       = var.subnets
   cidr_block     = each.value.cidr_block
   compartment_id = var.compartment_id
   vcn_id         = var.vcn_id
+  availability_domain = lookup(each.value, "availability_domain", null) != null ? data.oci_identity_availability_domains.ads.availability_domains[each.value.availability_domain - 1].name : null
```

Comment:
Thinking about this again, my understanding is that we recommend using regional subnets. Is there a reason you need to make it specific?

Also, if you still want to continue with this AD-specific direction, please look at https://github.com/oracle-terraform-modules/terraform-oci-oke/blob/main/module-iam.tf

Reply:
This request titled  _**Issue #115 - Allow availability domain configurable per subnet**_  was opened , if Oracle recommendation is to use regional subnet , should we have this request ?.

Reply:
It's still a valid use case, I was just wondering whether we should have it or not. I think some users might need it at scale.